src/main/java/nextstep/subway/domain/Sections.java

```diff
@@ -34,87 +35,104 @@ public void addInDown(Section newSection) {
                 .ifPresent(section -> section.updateDownStation(newSection.getUpStation(), newSection.getDistance()));
     }
 
+    public void mergeSection(Section targetSection) {
+        this.sections.stream()
+                .filter(section -> section.isDownStationEquals(targetSection.getUpStation()))
+                .findFirst()
+                .ifPresent(section -> section.updateDownStation(targetSection.getDownStation(), -targetSection.getDistance()));
+        removeSection(targetSection);
+    }
+
+    public void removeSection(Section section) {
+        if (isLeftOneSection()) {
+            throw new IllegalArgumentException("노선에 구간이 하나 남아 삭제할 수 없습니다.");
+        }
+        this.sections.remove(section);
+    }
+
+    private boolean isLeftOneSection() {
+        return sections.size() == 1;
+    }
+
     public boolean isEmpty() {
         return this.sections.isEmpty();
     }
 
-    public boolean isAddInUpSection(Section newSection) {
-        return getStations().get().contains(newSection.getUpStation());
+    public boolean exist(Station station) {
+        return sections.stream()
+                .anyMatch(section -> section.isUpStationEquals(station)
```

Comment:
```suggestion
                .anyMatch(section -> section.has(station));
```

이렇게 쓰면 좀더 간결하지 않을까요?